k3s-io/k3s
# Fix K3S_DATA_DIR when running as non-root user

#### Proposed Changes ####

When running the generated kill and uninstall scripts as non-root user, the environment variable `K3S_DATA_DIR` got lost, because `sudo` by default doesn't preserve the environment. This is fixed by adding the `--preserve-env` flag.

#### Types of Changes ####

Bugfix

#### Verification ####

* Set environment variable `K3S_DATA_DIR`. 
* Run the generated `k3s-uninstall.sh` script as non-root user and check the log output.
* Verify that the `$K3S_DATA_DIR` directory got cleaned. 

Fixes: #11386
